apps/discussions/contracts/DiscussionApp.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "@aragon/os/contracts/common/IForwarder.sol";
+
+
+contract DiscussionApp is IForwarder, AragonApp {
+    using SafeMath for uint256;
+
+    event Post(address indexed author, string postCid, uint discussionThreadId, uint postId, uint createdAt);
+    event Revise(address indexed author, string revisedPostCid, uint discussionThreadId, uint postId, uint createdAt, uint revisedAt);
+    event Hide(address indexed author, uint discussionThreadId, uint postId, uint hiddenAt);
+    event CreateDiscussionThread(uint actionId, bytes _evmScript);
+
+    bytes32 constant public DISCUSSION_POSTER_ROLE = keccak256("DISCUSSION_POSTER_ROLE");
```

Comment:
Having a single role for all three actions isn't exactly ideal, what I would consider doing is splitting hide into two function calls with a seperate `HIDE_DISCUSSION_ROLE` and then keeping the one that exists but removing the role protections. As it is in most cases the auth check will have occured at the time of post, so this pattern only serves to _prevent_ a poster from hiding their own post if they've had posting permissions removed. I think it makes more sense for `hide` to be something you can _always_ do for your own posts and a separate role can do for other peoples posts.